fission-web-server/library/Fission/Web/Server/Sentry.hs

```diff
@@ -6,57 +6,63 @@ module Fission.Web.Server.Sentry
   ) where
 
 import qualified RIO
-import qualified RIO.List                               as List
 import qualified RIO.Text                               as Text
 
+import           Data.Version                           (showVersion)
+
 import           System.Log.Raven
 import           System.Log.Raven.Transport.HttpConduit
 import           System.Log.Raven.Types                 as Sentry
 
+import qualified Servant.Client.Core                    as Servant
+
 import           Fission.Prelude                        hiding (onException)
 
+import           Fission.Web.Server.Host.Types
 import qualified Fission.Web.Server.Sentry.DSN.Types    as Sentry
 
+import qualified Paths_fission_web_server               as Fission
+
 -- | Instantiate a Sentry logger
-mkLogger :: MonadUnliftIO m => RIO.LogLevel -> Sentry.DSN -> m LogFunc
-mkLogger minRIOLogLevel (Sentry.DSN dsn) = do
+mkLogger :: MonadUnliftIO m => Host -> RIO.LogLevel -> Sentry.DSN -> m LogFunc
+mkLogger host minRIOLogLevel (Sentry.DSN dsn) = do
   raven <- liftIO $ initRaven dsn identity sendRecord silentFallback
   raven
-    |> logger minRIOLogLevel
+    |> logger host minRIOLogLevel
     |> mkLogFunc
     |> pure
 
 -- | Log from inside the application
 logger
   :: MonadUnliftIO m
-  => RIO.LogLevel
+  => Host
+  -> RIO.LogLevel
   -> SentryService
   -> CallStack
   -> LogSource
   -> RIO.LogLevel
   -> Utf8Builder
   -> m ()
-logger minRIOLogLevel sentryService _cs logSource logLevel msg =
+logger (Host host) minRIOLogLevel sentryService _cs _logSource logLevel msg =
   liftIO $ when (logLevel >= minRIOLogLevel) do
-    timestamp <- getCurrentTime
-    register sentryService loggerName level (message timestamp) (sentryRecord timestamp)
+    now <- getCurrentTime
+    register sentryService loggerName level message (sentryRecord now)
   where
     level :: SentryLevel
     level = fromRIOLogLevel logLevel
 
-    message :: UTCTime -> String
-    message time = mconcat <| List.intersperse " - - "
-      [ Text.unpack (textDisplay msg)
-      , show time
-      , show logLevel
-      , if logSource == mempty then "" else Text.unpack logSource
-      ]
+    message :: String
+    message = Text.unpack $ utf8BuilderToText msg
 
     sentryRecord :: UTCTime -> SentryRecord -> SentryRecord
     sentryRecord time oldRecord =
       oldRecord
-        { srMessage = message time
-        , srLevel   = level
+        { srMessage    = message
+        , srLevel      = level
+        , srTimestamp  = time
+        , srPlatform   = Just "Haskell/Servant"
+        , srServerName = Just $ Servant.baseUrlHost host
+        , srRelease    = Just $ showVersion Fission.version
```

Comment:
While I had the hood up, I needed to do some full-system integration debugging (browser, server, &c). So, hey, here's some at least slightly better Sentry logging